src/compat/db_vdb.c

```diff
@@ -658,37 +666,47 @@ db_compile_statement_local (DB_SESSION * session)
 	}
     }
 
-  /* translate views or virtual classes into base classes */
-  statement_result = mq_translate (parser, statement);
-  if (!statement_result || pt_has_error (parser))
+  if ((statement->node_type == PT_INSERT && statement->info.insert.spec->info.spec.remote_server_name)
+      || (statement->node_type == PT_DELETE && statement->info.delete_.spec->info.spec.remote_server_name)
+      || (statement->node_type == PT_UPDATE && statement->info.update.spec->info.spec.remote_server_name)
+      || (statement->node_type == PT_MERGE && statement->info.merge.into->info.spec.remote_server_name))
```

Comment:
I disagree with skipping mq_translate(). Various processing such as VIEW processing will not proceed.

Reply:
This is only case of remote DML. In general DML case, the mq_translate() is called.

Reply:
I disagree with skipping mq_translate() even for remote DML.

Reply:
mq_translate() is executed at remote server side. therefore the mq_tranlate at local side is not necessary.

Reply:
There may be cases like the following.
```
update remote@s1 a, local_view b 
     set ....
where a.col1 = b.col1
```

Reply:
In this version, this case results error because the remote DML query is not allow the join with a local view yet. The error is handled at pt_convert_dblink_dml_query.